scripts/watchPagesFolder.js

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+let lastLog = 0;
+
+// Debounce does not work
+function logGenerateDone() {
+  const now = Date.now();
+  if (now - lastLog > 1000) {
+    process.stdout.write('🤖 Route types have been updated\n');
+    lastLog = now;
+  }
+}
+
+function watchPagesFolder() {
+  return new Promise((resolve, reject) => {
+    const chokidar = require('chokidar');
+    const path = require('path');
+    const generateRoutesType = require('./generateRoutesType');
+
+    const watcher = chokidar.watch(path.resolve('src/pages'), {
+      ignored: /^\./,
+      persistent: true,
+    });
+
+    watcher.on('all', (event, path) => {
+      // Ignore file changes (file/folder name changes have the add/addDir event)
+      if (event !== 'change') {
+        try {
+          generateRoutesType(event, path);
+          logGenerateDone();
+        } catch (err) {
+          reject(err);
+        }
+      }
+    });
+
+    watcher.on('error', (err) => {
+      reject(err);
+    });
+  });
+}
+
+module.exports = watchPagesFolder;
```

Comment:
Can you explain this new scripts folder?

Reply:
In the past we used `next-routes` to create a routes object. We could then infer a `Routes` type from this object to make routing type-checking possible. I removed the `next-routes` package because it became obsolete due to Next adding dynamic routes support. This means we no longer have to update and keep track of a routes object, which also means we lose type-checking on routes.

This specific scripts creates a watcher for the `pages` folder and waits for updates on the folder structure. From this it will automatically create route types that you can see here https://github.com/react-prime/react-prime-ssr/blob/v10/src/types/routes.ts

Reply:
I have removed these auto-generating scripts for now. They require more testing in an actual project to see what edge cases come up.

Reply:
Ah okay, I was wondering why you'd need this besides HMR. The auto generating types explains :)